src/solver/solverUtils.F90

```diff
@@ -392,6 +392,272 @@ subroutine gridVelocitiesFineLevel(useOldCoor, t, sps)
   end subroutine gridVelocitiesFineLevel
 #endif
 
+
+#ifndef USE_TAPENADE
+  subroutine gridVelocitiesFineLevel_TS(sps)
+    !
+    ! Shell function to call gridVelocitiesFineLevel on all blocks
+    !
+    use blockPointers
+    use constants
+    use inputTimeSpectral
+    use iteration
+    use utils, only : setPointers
+    implicit none
+    !
+    !      Subroutine arguments.
+    !
+    integer(kind=intType), intent(in) :: sps
+    integer(kind=intType) :: nn
+
+    ! Loop over the number of blocks.
+
+    domains: do nn=1,nDom
+
+       ! Set the pointers for this block.
+
+       call setPointers(nn, groundLevel, sps)
+       call gridVelocitiesFineLevel_TS_block(nn, sps)
+
+    end do domains
+
+  end subroutine gridVelocitiesFineLevel_TS
+#endif
+
+  subroutine gridVelocitiesFineLevel_TS_block(nn, sps)
+
+    use precision
+    use constants
+    use blockPointers!, only: nDom, ie, je, ke, x, s, sFaceI, sI, sFaceJ, sJ, sFaceK, sK
+    use inputPhysics, only: machgrid, velDirFreestream
+    use flowVarRefState, only: gammaInf, pInf, rhoInf
+    !use partitioning, only: timePeriodSpectral
```

Comment:
Can some of these commented out lines be removed?

Reply:
Deleted